src/Mod/TechDraw/Gui/PagePrinter.cpp

```diff
@@ -230,6 +231,9 @@ void PagePrinter::printAll(QPrinter* printer, App::Document* doc)
         if (!vpp) {
             continue;// can't print this one
         }
+        // is there always a mdi when printAll is called?
+        auto mdi = vpp->getMDIViewPage();
+        mdi->savePageExportState(vpp);
```

Comment:
Here you need `mdi` only for class `m_docModStateBeforePrint` variable. Using local variable solves that issue.